From 58986a509da1c6387435b27c4071944ee14613a2 Mon Sep 17 00:00:00 2001
From: "Brian C. Lane" <bcl@redhat.com>
Date: Fri, 24 Feb 2017 11:06:20 -0800
Subject: [PATCH 76/77] Increase timeout for rmmod scsi_debug and make it a
 framework failure

On some hardware scsi_debug can take longer to remove. If this fails it
inevitably results in the next test failing since it cannot load
scsi_debug.

Increse the timeout to 40 loops (8 seconds), and if it hits the limit
without succeeding call framework_failure_

(cherry picked from commit 40e55a0819862d5dbe2098803735f8c48fb30b6d)
---
 tests/t-local.sh | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tests/t-local.sh b/tests/t-local.sh
index 13073d2..2ba7ee1 100644
--- a/tests/t-local.sh
+++ b/tests/t-local.sh
@@ -27,13 +27,14 @@ scsi_debug_cleanup_()
     # "Module scsi_debug is in use".
     i=0
     udevadm settle
-    while [ $i -lt 10 ] ; do
+    while [ $i -lt 40 ] ; do
       rmmod scsi_debug \
 	&& { test "$VERBOSE" = yes && warn_ $ME_ rmmod scsi_debug...; break; }
       sleep .2 || sleep 1
       i=$((i + 1))
     done
     udevadm settle
+    test $i = 40 && framework_failure_ rmmod scsi_debug failed.
   fi
   rm -fr $scsi_debug_lock_dir_
 }
-- 
2.9.3

